subsys/bluetooth/mesh/access.c

```diff
@@ -1678,7 +1678,7 @@ int bt_mesh_model_extend(struct bt_mesh_model *extending_mod, struct bt_mesh_mod
 	/* Check if a's list contains b */
 	for (it = a; (it != NULL) && (it->next != a); it = it->next) {
 		if (it == b) {
-			return 0;
+			goto register_extension;
```

Comment:
What will happen if the function is called for the same models a couple of times?
Will the CDP1 be overflowed with the same dependencies?

Reply:
I suppose it might be added multiple times, though the easiest solution would be to just not call it multiple times with the same parameters. 
Could however add some sort of check for this in mod_rel_register, or do you have any suggestions?

Reply:
I'd suggest to check presence of the registered data in CDP1. If they already exist, just leave CDP1 as is.

```
to just not call it multiple times with the same parameters
```
is not an obvious restriction. I'd prefer to have as many restrictions for a customer as possible. It will be more stable and user-friendly.

Reply:
This check sounds overcomplicated, doesn't it? Why would you call `bt_mesh_model_extend` or `bt_mesh_model_correspond` twice? The hierarchy of the models (at least MMDL) is defined so that need to declare relationship only once.

Reply:
This check is added IIRC to avoid creating circular linked list.

Reply:
If the hierarchy of the models is defined, how the circular linked list is possible?
Should this checking be removed at all?

Reply:
Ok, sorry, I explained this incorrectly. Multiple models can extend a single model, while 2 models never extend each other. The first happens in Generic Power Level and Power OnOff models:

1. plvl_setup_srv extends ponoff_setup_srv
2. plvl_setup_srv extends plvl_srv
3. plvl_srv extends ponoff_srv
4. ponoff_setup_srv extends ponoff_srv

So this check prevents creating a loop in point 4.On the other hand we should register this relation in CDP1.

What a user shall never do is to call `bt_mesh_model_extend` and `bt_mesh_model_correspond` for models A and B twice: (A, B) and with (B, A) as arguments.

Reply:
Ok, I'll approve this, but this is kind of hidden knowledge. If I do my own proprietary models (those are not according to mmdl), I know nothing about these limitations. Then available place in CDP1 will depend on how my app initializes the models. Kind of bug prone.